channeldb/channel.go

```diff
@@ -1360,6 +1360,51 @@ func (c *OpenChannel) SecondCommitmentPoint() (*btcec.PublicKey, error) {
 	return input.ComputeCommitmentPoint(revocation[:]), nil
 }
 
+// PersistDeliveryScript is used during the cooperative close flow to persist
+// a script sent in Shutdown when we did not set an upfront shutdown script
+// during the funding flow.
+func (c *OpenChannel) PersistDeliveryScript(
+	deliveryScript lnwire.DeliveryAddress) error {
+
+	c.Lock()
+	defer c.Unlock()
+
+	if err := kvdb.Update(c.Db.backend, func(tx kvdb.RwTx) error {
+		chanBucket, err := fetchChanBucketRw(
+			tx, c.IdentityPub, &c.FundingOutpoint, c.ChainHash,
+		)
+		if err != nil {
+			return err
+		}
+
+		channel, err := fetchOpenChannel(
+			chanBucket, &c.FundingOutpoint,
+		)
+		if err != nil {
+			return err
+		}
+
+		channel.LocalShutdownScript = deliveryScript
```

Comment:
What about also persisting the last offer we sent over? Np, in doing in another PR (should make an issue for).